app/ui.js

```diff
@@ -156,6 +156,38 @@ var UI = {
             }
         }
 
+        /*
+         * If novnc is hosted at PATHNAME, the default ws path should be at WS:
+         *  PATHNAME                  WS
+         *  /                         /websockify
+         *  /foo/myvnc                /foo/myvnc/websockify
```

Comment:
Do we really want this behaviour? I'd rather avoid getting to deep in to the mess of determining if this is a directory or a file.

Reply:
Anything that tries to guess if this is a folder or a file. Just assume everything after the last `/` should be removed. So `/foo/myvnc` => `/foo/websockify` and `/bar/vnc.html` => `/bar/websockify`.

Reply:
Okay, thanks. Should I treat /foo/myvnc/ the same as /foo/myvnc then? (Pedantically) that would go against "assume everything after the last `/` should be removed" since in the trailing slash case it would remove nothing. But I think it is more in-line with "avoid ... determining if this is a directory or a file".